lib/Fencer/Rules.hs

```diff
@@ -100,6 +118,26 @@ loadRulesFromDirectory
         dirs <- filterM isDirectory other
         (files ++) <$> concatMapM listAllFiles dirs
 
+-- | Perform validation checks to make sure the behavior matches that
+-- of @lyft/ratelimit@.
+validatePotentialDomains
+  :: [Either LoadRulesError (Maybe DomainDefinition)]
+  -> Either [LoadRulesError] [DomainDefinition]
+validatePotentialDomains res = case partitionEithers res of
+  (errs@(_:_), _    ) -> Left errs
```

Comment:
Maybe return a `Either (NonEmpty LoadRulesError) [DomainDefinition]` instead of `Either [LoadRulesError] [DomainDefinition]` then?

Reply:
Done! Thank you for the suggestion.